src/lexer.rs

```diff
@@ -160,7 +161,18 @@ impl<'a> Lexer<'a> {
                 token = self.char_token(TokenKind::Divide);
             }
             '.' => {
-                token = self.char_token(TokenKind::Period);
+                let p = self.peek_char();
+                if p.is_ascii_digit() {
+                    if let Some(t) = self.read_decimal_part(self.position) {
+                        token = t;
+                    }
+                    else {
```

Comment:
I'm unhappy about this, and need to look into the Rust `Option` mechanism to figure out how to have `read_decimal_part` always return a `Token`, but `read_number` return an `Option<Token<'a>>`. A simpler approach might be to have `read_number` always return a non-optional value, and force the integer parser to do junk cleanup too...

Reply:
This looks decent to me. Easy to follow.

Reply:
I'm unhappy about the repeated code in the inner and outer `else` blocks... Especially as the inner one should never be triggered because `read_decimal_part` always returns a `Token` in its `Option`.

I'll leave it for now, though. It may change as we evolve the lexer anyway.